File: util/data/src/main/java/dev/sweety/data/PrettyEnum.java
package dev.sweety.data;

public interface PrettyEnum {

    String name();

    default String prettyName() {
        return this.name().toLowerCase();
    }

    default String camelName() {
        String[] parts = this.name().toLowerCase().split("_");
        StringBuilder camelCaseName = new StringBuilder();
        for (String part : parts) {
            camelCaseName.append(part.substring(0, 1).toUpperCase()).append(part.substring(1));
        }
        return camelCaseName.toString();
    }

}

File: util/math/src/main/kotlin/dev/sweety/data/PrettyEnum.kt
package dev.sweety.data

interface PrettyEnum {

    fun prettyName(): String = (this as Enum<*>).name.lowercase()

    fun camelName(): String {
        val parts = prettyName().split("_")
        val camelCaseName = StringBuilder()
        for (part in parts) {
            camelCaseName.append(part.substring(0, 1).uppercase()).append(part.substring(1))
        }
        return camelCaseName.toString()
    }
}
